plugin.md

```diff
@@ -2,11 +2,42 @@
 
 ## Writing Plugins
 
+The main method that gets called is `ServeDNS` it has three parameters:
+
+* a `context.Context`
+* `dns.ResponseWriter` that is, basically, the client's connection.
+* `*dns.Msg` the request from the client.
+
```

Comment:
A nit, but I would put a semi-colon at the end of each of the first two bullets (instead of the period in the second one). The period is only on the last bullet.